dask/dataframe/shuffle.py

```diff
@@ -262,6 +270,24 @@ def shuffle(df, index, shuffle=None, npartitions=None, max_branch=32, compute=No
     set_partition
     shuffle_disk
     """
+    if shuffle == "tasks" and isinstance(index, (str, list)):
```

Comment:
`index` is coming straight from users, so we should be a bit more forgiving on the list type. Perhaps use `pandas.api.types.is_list_like(index)`.

Reply:
Good point - I am now allowing the index to be "list_like", as long as it is not a dask collection.